src/main/java/com/kainos/ea/controller/WebService.java

```diff
@@ -38,6 +61,125 @@ public Response getJobRoles()
+        }
+    }
+
+    @GET
+    @Path("/bands")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getBands(){
+        try{
+            return Response.status(HttpStatus.OK_200).entity(bandsService.getAllBands()).build();
+        } catch (Exception | DatabaseConnectionException e) {
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        }
+    }
+
+    @GET
+    @Path("/job-families/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getFamilyById(@PathParam("id") int id){
+        try {
+            return Response.status(HttpStatus.OK_200).entity(familiesService.getFamilyById(id)).build();
+        } catch (Exception | DatabaseConnectionException  e){
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        } catch (FamilyDoesNotExistException e){
+            return Response.status(HttpStatus.NOT_FOUND_404, e.getMessage()).build();
+        }
+    }
+
+    @GET
+    @Path("/job-families")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getFamilies(){
+        try{
+            return Response.status(HttpStatus.OK_200).entity(familiesService.getAllFamilies()).build();
+        } catch (Exception | DatabaseConnectionException e) {
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        }
+    }
+
+    @PUT
+    @Path("/job-families")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateJobFamily(JobFamily jobFamily){
+        try {
+            int id = familiesService.updateJobFamily(jobFamily);
+            return Response.status(HttpStatus.OK_200).entity(id).build();
+        } catch (Exception | DatabaseConnectionException e) {
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+        }
+    }
+
+    @GET
+    @Path("/capabilities/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getCapabilityById(@PathParam("id") int id){
+        try {
+            return Response.status(HttpStatus.OK_200).entity(capabilitiesService.getCapabilityById(id)).build();
+        } catch (Exception | DatabaseConnectionException  e){
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        } catch (CapabilityDoesNotExistException e){
+            return Response.status(HttpStatus.NOT_FOUND_404, e.getMessage()).build();
+        }
+    }
+
+    @GET
+    @Path("/capabilities")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getCapabilities(){
+        try{
+            return Response.status(HttpStatus.OK_200).entity(capabilitiesService.getAllCapabilities()).build();
+        } catch (Exception | DatabaseConnectionException e) {
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        }
+    }
```

Comment:
not sure why this endpoint is needed for your story?

Reply:
removed